@here/harp-mapview/lib/MapView.ts

```diff
@@ -2438,13 +2464,14 @@ export class MapView extends THREE.EventDispatcher {
 
         const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
         const cameraPosZ = this.getCameraHeightAboveTerrain(TERRAIN_ZOOM_LEVEL);
+        const zoomLevelDistance = cameraPosZ / Math.cos(Math.min(cameraPitch, Math.PI / 3));
+        this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(zoomLevelDistance, this);
+        this.m_fog.update(this, this.m_viewRanges.maximum);
 
         const target = MapViewUtils.rayCastWorldCoordinates(this, 0, 0);
```

Comment:
Instead of raycasting we should compute the target as `camera.position + camera.view * zoomLevelDistance' 

That would at least ensure consistence (i.e. distance between target and camera is always equal to `zoomLevelDistance`

Reply:
> Instead of raycasting we should compute the target as `camera.position + camera.view * zoomLevelDistance'
> 
> That would at least ensure consistence (i.e. distance between target and camera is always equal to `zoomLevelDistance`

I am afraid that this is not true for globe projection, this simple trigonometry can not be applied here,
cos(pitch) = altitude / zoomLevelDistance,
zoomLevelDistance = altitude / cos(pitch)
This comes from right (angle) triangle formed by the camera position, camera position projected on the ground and target point on the ground.
Unfortunately when we consider globe geometry, we do not have flat ground anymore, so by using this simplification our target point (rotation pivot) will be floating above the ground.
I am not sure if we want this, aren't you?

Reply:
Yeah, I was also thinking about that and wondering if we can have a similar formula for globe that would not rely on raycasting but just the camera position and camera pitch.
In that case we could just clamp the pitch and always compute the distance and with that distance we could compute "some" target point that might not match the real target point.

That would happen whenever the raycasting would also fail but that's fine b/c you can only get into this situation when you modify the camera directly. We just have to make sure that we can still compute everything we need internally (e.g. zoomlevel, near/far plane, ...)